src/main/java/com/google/sps/servlets/FamilyServlet.java

```diff
@@ -26,28 +26,42 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet responsible for creating new tasks. */
+/** Servlet responsible for creating new families. */
 @WebServlet("/family")
 public class FamilyServlet extends HttpServlet {
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         UserService userService = UserServiceFactory.getUserService();
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        String userEmail = userService.getCurrentUser().getEmail();
+        Query query = new Query("UserInfo")
+            .setFilter(new Query.FilterPredicate("email", Query.FilterOperator.EQUAL, userEmail));
+        PreparedQuery results = datastore.prepare(query);
+        Entity entity = results.asSingleEntity();
+        if (entity != null) {
+            System.out.println("You already belong to a family");
```

Comment:
This adds the message to the stdout logs (which I'm not sure is actually saved for GAE). The user will see no indication that anything went wrong (I think they may just see the form again).

Let's talk about how to do error handling tomorrow after the meeting.

https://docstore.mik.ua/orelly/java-ent/servlet/ch05_07.htm for some reading.